howto_io.md

```diff
@@ -0,0 +1,268 @@
+# How-to: NumPy I/O

```

Comment:
Needs a two sentence intro

Reply:
Thanks for taking time to review this.

> Is there a reason you chose not to present this as a Jupyter Notebook?

* It was a prototype to see if there was support for the PR.
* Unlike a tutorial, where the user is invited to work with the examples, this are rote steps; many are just links.
* To make this executable in a user environment (for instance, via binder), files used in the examples would be introduced as dependencies, plus libraries like Zarr that the user would need to install. 
* If it's just documentation and not user-runnable, nothing is added by making it a notebook; Markdown is easier to update and review.

> The file itself should be moved to the content directory

Since it isn't a notebook, I wanted to keep it clear of whatever CI / build mechanism is going on in the repo till I knew what would happen if an .md were thrown in.

Also the repo isn't yet organized to separately handle tutorials and how-tos.

> Maybe the IO section of this repo should have two documents: a tutorial and a how-to. PR gh-14 is more of a tutorial, this is more of a how-to.

I agree there's value in covering topics with both tutorials and how-tos.  gh-14 has a tutorial feel but doesn't qualify as a tutorial in its current form. A tutorial needs a single narrative thread; it can't be a catalog.

> Needs a two sentence intro
If the title is something like "How to: NumPy I/O", what would be added by an intro?

Reply:
It would set expectations and provide keywords to help discoverability. "NumPy has many methods to serialize ndarrays. This How-To will provide an overview of when to choose which one, based on the task at hand, with links for more information and examples. Since this is a How-To (link to the how-to explaination), it is purposefully abbreviated to be as short as possible."

Reply:
Opening with "NumPy has many methods to serialize ndarrays" is like answering "There are many ways to tell time" to someone who asks the time.

Just like the guy asking for the time, the reader's expectations are already in place: they have a how-to question and expect to find the answer. Our job is to meet the expectations or fail.

If it weren't obvious how to use the contents -- if, say, we were presenting a triangular mileage map -- we might want to spare a few words to explain it. But everyone can pattern-match on headings. 

![image](https://user-images.githubusercontent.com/6691888/89124640-918fc700-d4a6-11ea-917f-fc5162a43ca8.png)

Reply:
The difference is that we are not the only person for miles around on a rural road being queried nor are we an atlas sitting on a dusty shelf. We are a  web-page on an internet full of other versions of the same information. If we wish to be noticed, we need to rise to the top of the heap: we need to be indexed and searchable and appear at the top of searches. An appropriate minimum of words helps that process. A bare list of links does not: it scores very low and will not be noticed.

Reply:
I might agree more if I could find a single useful keyword in:

> NumPy has many methods to serialize ndarrays. This How-To will provide an overview of when to choose which one, based on the task at hand, with links for more information and examples. Since this is a How-To (link to the how-to explaination), it is purposefully abbreviated to be as short as possible.

The "bare links" point to headings which are full of keywords.  Search engines have no trouble finding our api pages, which have no empty sentences propping them up.